File: client/src/app/models/classParams.ts
import { PaginationParams } from './paginationParams';
export class ClassParams extends PaginationParams{
    studentId?: number;

    constructor(public beginDate: Date = null,
                public endDate: Date = null,
                public levelId: number = 0,
                public teacherId: number = 0)
    {
        super();
        if (!this.beginDate)
        {
            this.beginDate = new Date();
            this.beginDate.setDate((new Date()).getDate() - 7);
        }

        if (!endDate)
        {
            this.endDate = new Date();
            this.endDate.setDate((new Date()).getDate() + 7);
        }

        if (this.levelId === 0){
            this.levelId = null;
        }

        if (this.teacherId === 0){
            this.teacherId = null;
        }
    }
}

File: API/Helpers/ClassParams.cs
using System;

namespace API.Helpers
{
    public class ClassParams: PaginationParams
    {
        public DateTime BeginDate { get; set; } = DateTime.Today;
        public DateTime EndDate { get; set; } = DateTime.Today.AddDays(7);
        public int? LevelId { get; set; }
        public int? StudentId { get; set; }
        public int? TeacherId { get; set; }
        public bool Suspended { get; set; }
    }
}